docs/advanced/extensibleenums.md

```diff
@@ -7,216 +7,134 @@ Extensible Enums are an enhancement of specific Vanilla enums to allow new entri
-    }
-
-    // Matches the constructor
-    public static ExampleEnumInit create(String name, int arg1, String arg2, boolean arg3) {
-        throw new IllegalStateException("Enum not extended");
-    }
-
-    @Override
-    public void init() {
-        // Gets called for every new enum
-        ARG2_TO_ARG3.put(this.arg2, this.arg3);
+public class MyEnumParams {
+    public static final EnumProxy<Rarity> CUSTOM_RARITY_ENUM_PROXY = new EnumProxy<>(
+            Rarity.class, -1, "examplemod:custom", (UnaryOperator<Style>) style -> style.withItalic(true)
+    );
+    
+    public static Object getTestDamageEffectsParameter(int idx, Class<?> type) {
+        return type.cast(switch (idx) {
+            case 0 -> "examplemod:test";
+            case 1 -> (Supplier<SoundEvent>) () -> SoundEvents.DONKEY_ANGRY;
+            default -> throw new IllegalArgumentException("Unexpected parameter index: " + idx);
+        });
     }
 }
 ```
 
-### Codecs
+#### Constructor
 
-[Enum codecs][codec] generally take in the `values` array and resolve the codec instantly. As the array is evaluated before any mod entries are added, no mod entries will be supported. There are two solutions to this problem, which to choose depends on whether the enum implements the `StringRepresentable` interface or not.
+The constructor must be specified as a method descriptor and must only contain the parameters visible in the source code, omitting the hidden constant name and ordinal parameters
 
-For normal, non-`StringRepresentable` enums, the codec can be wrapped via `Codec#lazyInitialized` or [`NeoForgeStreamCodecs#lazy`][streamcodec]. These prevent the codec from being resolved until first usage, which will always be after all mod entries are added.
+#### Parameters
 
-```java
-// For some enum with codec
-public enum ExampleEnumCodec {
-    // ...
-    ;
+The parameters can be specified in three ways with limitations depending on the parameter types:
 
-    public static final Codec<ExampleEnumCodec> CODEC = Codec.of(/* ... */);
-    public static final StreamCodec<ByteBuf, ExampleEnumCodec> CODEC = StreamCodec.of(/* ... */);
+- Inline in the JSON file as an array of constants (only allowed for primitive values, Strings and for passing null to any reference type)
+- As a reference to a field of type `EnumProxy<TheEnum>` in a class from the mod
+- As a reference to a method taking a parameter index and a `Class` object as parameters and returning `Object`
+  - The `Class` object is the type expected for the given parameter and should be used to cast (`Class#cast()`) the return value in order to keep `ClassCastException`s in mod code
 
-    // ...
-}
-
-// The final extended enum may look something like this
-public enum ExampleEnumCodec implements net.neoforged.neoforge.common.IExtensibleEnum {
-    // ...
-    ;
+:::warning
+The fields and/or methods used as parameter sources should be in a separate class to avoid unintentionally loading mod classes too early
+:::
```

Comment:
```suggestion
:::tip
The fields and/or methods used as sources for parameter values should be in a separate class to avoid unintentionally loading mod classes too early.
:::
```

Reply:
I disagree with making this a tip, it's not merely a nice-to-have.